File: android/app/src/main/java/co/inboxies/app/ui/settings/DomainAdminSettingsView.kt
package co.inboxies.app.ui.settings

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.outlined.ArrowBack
import androidx.compose.material.icons.outlined.Refresh
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.RadioButton
import androidx.compose.material3.RadioButtonDefaults
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalClipboardManager
import androidx.compose.ui.text.AnnotatedString
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import co.inboxies.app.LocalAppModel
import co.inboxies.app.models.AdminMailboxRow
import co.inboxies.app.services.ApiClient
import co.inboxies.app.theme.HomeChromeToolbarButton
import co.inboxies.app.theme.InterFontFamily
import co.inboxies.app.theme.inboxiesColors
import kotlinx.coroutines.launch

@Composable
fun DomainAdminSettingsView(
    onBack: (() -> Unit)? = null,
    onAssigned: (() -> Unit)? = null,
) {
    val app = LocalAppModel.current
    val colors = inboxiesColors()
    val scope = rememberCoroutineScope()
    val clipboard = LocalClipboardManager.current

    var rows by remember { mutableStateOf<List<AdminMailboxRow>>(emptyList()) }
    var loading by remember { mutableStateOf(true) }
    var error by remember { mutableStateOf<String?>(null) }
    var status by remember { mutableStateOf<String?>(null) }
    var lastInviteUrl by remember { mutableStateOf<String?>(null) }
    var assigningId by remember { mutableStateOf<String?>(null) }
    var showCreate by remember { mutableStateOf(false) }
    var inviteMailboxId by remember { mutableStateOf<String?>(null) }
    var deleteTarget by remember { mutableStateOf<AdminMailboxRow?>(null) }

    fun reload() {
        scope.launch {
            loading = rows.isEmpty()
            error = null
            runCatching { ApiClient.shared.listAdminMailboxes() }
                .onSuccess { rows = it }
                .onFailure { error = it.message }
            loading = false
        }
    }

    LaunchedEffect(Unit) { reload() }

    Column(modifier = Modifier.fillMaxWidth()) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 8.dp, vertical = 4.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            if (onBack != null) {
                HomeChromeToolbarButton(
                    icon = Icons.AutoMirrored.Outlined.ArrowBack,
                    contentDescription = "Back",
                    onClick = onBack,
                )
            }
            Text(
                "Admin",
                fontFamily = InterFontFamily,
                fontWeight = FontWeight.SemiBold,
                fontSize = 17.sp,
                color = colors.ink,
                modifier = Modifier.weight(1f).padding(horizontal = 8.dp),
            )
            HomeChromeToolbarButton(
                icon = Icons.Outlined.Refresh,
                contentDescription = "Refresh",
                onClick = { reload() },
            )
        }

        Column(
            modifier = Modifier
                .fillMaxWidth()
                .verticalScroll(rememberScrollState())
                .padding(horizontal = 16.dp)
                .padding(bottom = 40.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp),
        ) {
            error?.let {
                Text(it, color = colors.deepDarkRed, fontFamily = InterFontFamily, fontSize = 13.sp)
            }
            status?.let {
                Text(it, color = colors.muted, fontFamily = InterFontFamily, fontSize = 13.sp)
            }
            lastInviteUrl?.let { url ->
                Text("Invite link", fontFamily = InterFontFamily, fontWeight = FontWeight.SemiBold, color = colors.ink)
                Text(url, fontFamily = InterFontFamily, fontSize = 11.sp, color = colors.muted)
                TextButton(onClick = {
                    clipboard.setText(AnnotatedString(url))
                    status = "Invite link copied"
                }) {
                    Text("Copy link", color = colors.accent, fontFamily = InterFontFamily)
                }
            }

            Button(
                onClick = { showCreate = true },
                colors = ButtonDefaults.buttonColors(containerColor = colors.ink, contentColor = colors.surface),
                shape = RoundedCornerShape(12.dp),
                modifier = Modifier.fillMaxWidth(),
            ) {
                Text("Create email", fontFamily = InterFontFamily, fontWeight = FontWeight.Medium)
            }

            Text(
                "Domain mailboxes",
                fontFamily = InterFontFamily,
                fontWeight = FontWeight.SemiBold,
                fontSize = 14.sp,
                color = colors.ink,
                modifier = Modifier.padding(top = 8.dp),
            )

            if (loading) {
                CircularProgressIndicator(modifier = Modifier.padding(24.dp))
            } else if (rows.isEmpty()) {
                Text(
                    "No mailboxes yet. Create the first address.",
                    fontFamily = InterFontFamily,
                    color = colors.muted,
                    fontSize = 14.sp,
                )
            } else {
                rows.forEach { row ->
                    Column(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(vertical = 8.dp),
                        verticalArrangement = Arrangement.spacedBy(6.dp),
                    ) {
                        Text(row.email, fontFamily = InterFontFamily, fontWeight = FontWeight.Medium, color = colors.ink)
                        Text(
                            if (row.claimed) "Claimed · ${row.name}" else "Unclaimed · ${row.name}",
                            fontFamily = InterFontFamily,
                            fontSize = 12.sp,
                            color = colors.muted,
                        )
                        Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                            TextButton(
                                onClick = {
                                    scope.launch {
                                        assigningId = row.id
                                        runCatching {
                                            ApiClient.shared.assignAdminMailboxToSelf(row.id)
                                            app.refreshMailboxes(showLoading = true)
                                            status = "Assigned to you"
                                            onAssigned?.invoke()
                                            reload()
                                        }.onFailure { error = it.message }
                                        assigningId = null
                                    }
                                },
                                enabled = assigningId == null,
                            ) {
                                Text("Assign to me", color = colors.accent, fontFamily = InterFontFamily)
                            }
                            TextButton(onClick = { inviteMailboxId = row.id }) {
                                Text("Invite", color = colors.accent, fontFamily = InterFontFamily)
                            }
                            TextButton(onClick = { deleteTarget = row }) {
                                Text("Delete", color = colors.deepDarkRed, fontFamily = InterFontFamily)
                            }
                        }
                    }
                }
            }
        }
    }

    if (showCreate) {
        DomainAdminCreateDialog(
            onDismiss = { showCreate = false },
            onCreated = { url ->
                showCreate = false
                lastInviteUrl = url
                status = if (url == null) "Mailbox created" else "Mailbox created — invite ready"
                reload()
            },
        )
    }

    inviteMailboxId?.let { mailboxId ->
        DomainAdminInviteDialog(
            mailboxId = mailboxId,
            onDismiss = { inviteMailboxId = null },
            onInvited = { url ->
                inviteMailboxId = null
                lastInviteUrl = url
                status = "Invite ready"
            },
        )
    }

    deleteTarget?.let { target ->
        AlertDialog(
            onDismissRequest = { deleteTarget = null },
            title = { Text("Delete mailbox", fontFamily = InterFontFamily) },
            text = {
                Text("Delete ${target.email}? This cannot be undone.", fontFamily = InterFontFamily)
            },
            confirmButton = {
                TextButton(onClick = {
                    scope.launch {
                        runCatching {
                            ApiClient.shared.deleteAdminMailbox(target.id)
                            app.refreshMailboxes(showLoading = true)
                            status = "Mailbox deleted"
                            reload()
                        }.onFailure { error = it.message }
                        deleteTarget = null
                    }
                }) {
                    Text("Delete", color = colors.deepDarkRed, fontFamily = InterFontFamily)
                }
            },
            dismissButton = {
                TextButton(onClick = { deleteTarget = null }) {
                    Text("Cancel", fontFamily = InterFontFamily)
                }
            },
        )
    }
}

@Composable
private fun DomainAdminCreateDialog(
    onDismiss: () -> Unit,
    onCreated: (String?) -> Unit,
) {
    val colors = inboxiesColors()
    val scope = rememberCoroutineScope()
    var localPart by remember { mutableStateOf("") }
    var displayName by remember { mutableStateOf("") }
    var assignSelf by remember { mutableStateOf(true) }
    var inviteEmail by remember { mutableStateOf("") }
    var inviteName by remember { mutableStateOf("") }
    var saving by remember { mutableStateOf(false) }
    var error by remember { mutableStateOf<String?>(null) }
    val app = LocalAppModel.current
    val domain by app.mailDomain.collectAsState()

    AlertDialog(
        onDismissRequest = onDismiss,
        title = { Text("Create email", fontFamily = InterFontFamily, fontWeight = FontWeight.SemiBold) },
        text = {
            Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
                OutlinedTextField(
                    value = localPart,
                    onValueChange = { localPart = it.substringBefore('@') },
                    label = { Text("Username") },
                    trailingIcon = { Text("@$domain", color = colors.muted, fontSize = 12.sp) },
                    singleLine = true,
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(10.dp),
                )
                OutlinedTextField(
                    value = displayName,
                    onValueChange = { displayName = it },
                    label = { Text("Display name (optional)") },
                    singleLine = true,
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(10.dp),
                )
                Row(verticalAlignment = Alignment.CenterVertically) {
                    RadioButton(
                        selected = assignSelf,
                        onClick = { assignSelf = true },
                        colors = RadioButtonDefaults.colors(selectedColor = colors.accent),
                    )
                    Text("Assign to me", fontFamily = InterFontFamily, color = colors.ink)
                }
                Row(verticalAlignment = Alignment.CenterVertically) {
                    RadioButton(
                        selected = !assignSelf,
                        onClick = { assignSelf = false },
                        colors = RadioButtonDefaults.colors(selectedColor = colors.accent),
                    )
                    Text("Invite someone", fontFamily = InterFontFamily, color = colors.ink)
                }
                if (!assignSelf) {
                    OutlinedTextField(
                        value = inviteEmail,
                        onValueChange = { inviteEmail = it },
                        label = { Text("Invitee email") },
                        singleLine = true,
                        modifier = Modifier.fillMaxWidth(),
                        shape = RoundedCornerShape(10.dp),
                    )
                    OutlinedTextField(
                        value = inviteName,
                        onValueChange = { inviteName = it },
                        label = { Text("Invitee name (optional)") },
                        singleLine = true,
                        modifier = Modifier.fillMaxWidth(),
                        shape = RoundedCornerShape(10.dp),
                    )
                }
                error?.let { Text(it, color = colors.deepDarkRed, fontSize = 13.sp) }
            }
        },
        confirmButton = {
            TextButton(
                enabled = !saving && localPart.isNotBlank() && (assignSelf || inviteEmail.isNotBlank()),
                onClick = {
                    scope.launch {
                        saving = true
                        error = null
                        runCatching {
                            ApiClient.shared.createAdminMailbox(
                                email = "$localPart@$domain",
                                name = displayName.ifBlank { localPart },
                                assignToSelf = assignSelf,
                                inviteEmail = inviteEmail.ifBlank { null },
                                inviteeName = inviteName.ifBlank { null },
                            )
                        }.onSuccess { onCreated(it.invite?.inviteUrl) }
                            .onFailure { error = it.message }
                        saving = false
                    }
                },
            ) {
                Text(if (saving) "Creating…" else "Create", color = colors.accent, fontFamily = InterFontFamily)
            }
        },
        dismissButton = {
            TextButton(onClick = onDismiss) {
                Text("Cancel", fontFamily = InterFontFamily)
            }
        },
    )
}

@Composable
private fun DomainAdminInviteDialog(
    mailboxId: String,
    onDismiss: () -> Unit,
    onInvited: (String) -> Unit,
) {
    val colors = inboxiesColors()
    val scope = rememberCoroutineScope()
    var inviteEmail by remember { mutableStateOf("") }
    var inviteName by remember { mutableStateOf("") }
    var asOwner by remember { mutableStateOf(true) }
    var saving by remember { mutableStateOf(false) }
    var error by remember { mutableStateOf<String?>(null) }

    AlertDialog(
        onDismissRequest = onDismiss,
        title = { Text("Invite", fontFamily = InterFontFamily, fontWeight = FontWeight.SemiBold) },
        text = {
            Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
                Text(mailboxId, fontFamily = InterFontFamily, color = colors.muted, fontSize = 13.sp)
                OutlinedTextField(
                    value = inviteEmail,
                    onValueChange = { inviteEmail = it },
                    label = { Text("Invitee email") },
                    singleLine = true,
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(10.dp),
                )
                OutlinedTextField(
                    value = inviteName,
                    onValueChange = { inviteName = it },
                    label = { Text("Name (optional)") },
                    singleLine = true,
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(10.dp),
                )
                Row(verticalAlignment = Alignment.CenterVertically) {
                    RadioButton(
                        selected = asOwner,
                        onClick = { asOwner = true },
                        colors = RadioButtonDefaults.colors(selectedColor = colors.accent),
                    )
                    Text("Owner", fontFamily = InterFontFamily)
                    RadioButton(
                        selected = !asOwner,
                        onClick = { asOwner = false },
                        colors = RadioButtonDefaults.colors(selectedColor = colors.accent),
                    )
                    Text("Member", fontFamily = InterFontFamily)
                }
                error?.let { Text(it, color = colors.deepDarkRed, fontSize = 13.sp) }
            }
        },
        confirmButton = {
            TextButton(
                enabled = !saving && inviteEmail.isNotBlank(),
                onClick = {
                    scope.launch {
                        saving = true
                        runCatching {
                            ApiClient.shared.createAdminInvite(
                                mailboxId,
                                inviteEmail.trim(),
                                inviteName.ifBlank { null },
                                if (asOwner) "owner" else "member",
                            )
                        }.onSuccess { onInvited(it.inviteUrl) }
                            .onFailure { error = it.message }
                        saving = false
                    }
                },
            ) {
                Text(if (saving) "Sending…" else "Send", color = colors.accent, fontFamily = InterFontFamily)
            }
        },
        dismissButton = {
            TextButton(onClick = onDismiss) {
                Text("Cancel", fontFamily = InterFontFamily)
            }
        },
    )
}

File: ios/AgenticInbox/Inboxies/Views/Settings/DomainAdminSettingsView.swift
import SwiftUI

/// Domain Admin console — list/create/assign/invite/delete mailboxes.
struct DomainAdminSettingsView: View {
    @Environment(AppModel.self) private var app
    @Environment(\.dismiss) private var dismiss

    var showsDismiss: Bool = true
    var onAssigned: (() -> Void)? = nil

    @State private var rows: [AdminMailboxRow] = []
    @State private var isLoading = true
    @State private var errorMessage: String?
    @State private var assigningId: String?
    @State private var deletingId: String?
    @State private var showCreate = false
    @State private var inviteForId: String?
    @State private var lastInviteUrl: String?
    @State private var statusMessage: String?

    var body: some View {
        Group {
            if isLoading {
                ProgressView()
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
            } else {
                List {
                    if let errorMessage {
                        Section {
                            Text(errorMessage)
                                .font(.inter(size: 13))
                                .foregroundStyle(AppTheme.deepDarkRed)
                        }
                    }
                    if let statusMessage {
                        Section {
                            Text(statusMessage)
                                .font(.inter(size: 13))
                                .foregroundStyle(AppTheme.muted)
                        }
                    }
                    if let lastInviteUrl {
                        Section("Invite link") {
                            Text(lastInviteUrl)
                                .font(.system(size: 11, design: .monospaced))
                                .foregroundStyle(AppTheme.muted)
                                .textSelection(.enabled)
                            Button("Copy link") {
                                UIPasteboard.general.string = lastInviteUrl
                                statusMessage = "Invite link copied"
                            }
                            .font(.inter(size: 15, weight: .medium))
                            .foregroundStyle(AppTheme.accent)
                        }
                    }

                    Section {
                        Button {
                            showCreate = true
                        } label: {
                            Label("Create email", systemImage: "plus")
                                .font(.inter(size: 16, weight: .medium))
                                .foregroundStyle(AppTheme.accent)
                        }
                    }

                    Section("Domain mailboxes") {
                        if rows.isEmpty {
                            Text("No mailboxes yet. Create the first address.")
                                .font(.inter(size: 14))
                                .foregroundStyle(AppTheme.muted)
                        }
                        ForEach(rows) { row in
                            VStack(alignment: .leading, spacing: 8) {
                                Text(row.email)
                                    .font(.inter(size: 15, weight: .medium))
                                    .foregroundStyle(AppTheme.ink)
                                Text(row.claimed == true ? "Claimed · \(row.name)" : "Unclaimed · \(row.name)")
                                    .font(.inter(size: 12))
                                    .foregroundStyle(AppTheme.muted)
                                HStack(spacing: 16) {
                                    Button("Assign to me") {
                                        Task { await assignSelf(row.id) }
                                    }
                                    .disabled(assigningId != nil)
                                    Button("Invite") {
                                        inviteForId = row.id
                                    }
                                    Button("Delete", role: .destructive) {
                                        Task { await deleteMailbox(row.id) }
                                    }
                                    .disabled(deletingId != nil)
                                }
                                .font(.inter(size: 14, weight: .medium))
                            }
                            .padding(.vertical, 4)
                        }
                    }
                }
                .listStyle(.insetGrouped)
                .scrollContentBackground(.hidden)
            }
        }
        .background(AppTheme.background)
        .navigationTitle("Admin")
        .navigationBarTitleDisplayMode(.inline)
        .toolbar {
            if showsDismiss {
                ToolbarItem(placement: .topBarLeading) {
                    Button {
                        dismiss()
                    } label: {
                        Image(systemName: "chevron.left")
                            .font(.inter(size: 14, weight: .semibold))
                            .foregroundStyle(AppTheme.ink)
                            .frame(width: 32, height: 32)
                    }
                }
            }
            ToolbarItem(placement: .topBarTrailing) {
                Button {
                    Task { await reload() }
                } label: {
                    Image(systemName: "arrow.clockwise")
                        .foregroundStyle(AppTheme.ink)
                }
            }
        }
        .task { await reload() }
        .sheet(isPresented: $showCreate) {
            NavigationStack {
                DomainAdminCreateView { url in
                    lastInviteUrl = url
                    statusMessage = url == nil ? "Mailbox created" : "Mailbox created — invite ready"
                    Task { await reload() }
                }
            }
        }
        .sheet(item: Binding(
            get: { inviteForId.map { InviteTarget(id: $0) } },
            set: { inviteForId = $0?.id }
        )) { target in
            NavigationStack {
                DomainAdminInviteView(mailboxId: target.id) { url in
                    lastInviteUrl = url
                    statusMessage = "Invite ready"
                }
            }
        }
    }

    private func reload() async {
        isLoading = rows.isEmpty
        errorMessage = nil
        do {
            rows = try await APIClient.shared.listAdminMailboxes()
        } catch {
            errorMessage = error.localizedDescription
        }
        isLoading = false
    }

    private func assignSelf(_ mailboxId: String) async {
        assigningId = mailboxId
        defer { assigningId = nil }
        do {
            _ = try await APIClient.shared.assignAdminMailboxToSelf(mailboxId: mailboxId)
            await app.refreshMailboxes(showLoading: true)
            statusMessage = "Assigned to you"
            onAssigned?()
            await reload()
        } catch {
            errorMessage = error.localizedDescription
        }
    }

    private func deleteMailbox(_ mailboxId: String) async {
        deletingId = mailboxId
        defer { deletingId = nil }
        do {
            try await APIClient.shared.deleteAdminMailbox(mailboxId: mailboxId)
            if app.selectedMailboxId == mailboxId {
                await app.refreshMailboxes(showLoading: true)
            }
            statusMessage = "Mailbox deleted"
            await reload()
        } catch {
            errorMessage = error.localizedDescription
        }
    }
}

private struct InviteTarget: Identifiable {
    let id: String
}

private struct DomainAdminCreateView: View {
    @Environment(\.dismiss) private var dismiss
    var onDone: (String?) -> Void

    @State private var localPart = ""
    @State private var displayName = ""
    @State private var assignMode = 0 // 0 self, 1 invite
    @State private var inviteEmail = ""
    @State private var inviteName = ""
    @State private var isSaving = false
    @State private var errorMessage: String?

    private let domain = "inboxies.email"

    var body: some View {
        Form {
            Section {
                HStack {
                    TextField("username", text: $localPart)
                        .textInputAutocapitalization(.never)
                        .autocorrectionDisabled()
                        .onChange(of: localPart) { _, v in
                            if let at = v.firstIndex(of: "@") {
                                localPart = String(v[..<at])
                            }
                        }
                    Text("@\(domain)")
                        .foregroundStyle(AppTheme.muted)
                }
                TextField("Display name (optional)", text: $displayName)
            }
            Section("Assign to") {
                Picker("Assign", selection: $assignMode) {
                    Text("Me").tag(0)
                    Text("Invite someone").tag(1)
                }
                .pickerStyle(.segmented)
                if assignMode == 1 {
                    TextField("Invitee email", text: $inviteEmail)
                        .textInputAutocapitalization(.never)
                        .keyboardType(.emailAddress)
                    TextField("Invitee name (optional)", text: $inviteName)
                }
            }
            if let errorMessage {
                Section {
                    Text(errorMessage).foregroundStyle(AppTheme.deepDarkRed)
                }
            }
        }
        .navigationTitle("Create email")
        .navigationBarTitleDisplayMode(.inline)
        .toolbar {
            ToolbarItem(placement: .cancellationAction) {
                Button("Cancel") { dismiss() }
            }
            ToolbarItem(placement: .confirmationAction) {
                Button("Create") {
                    Task { await create() }
                }
                .disabled(isSaving || localPart.isEmpty || (assignMode == 1 && inviteEmail.isEmpty))
            }
        }
    }

    private func create() async {
        isSaving = true
        defer { isSaving = false }
        errorMessage = nil
        let email = "\(localPart)@\(domain)"
        do {
            let result = try await APIClient.shared.createAdminMailbox(
                email: email,
                name: displayName.isEmpty ? localPart : displayName,
                assignToSelf: assignMode == 0,
                inviteEmail: assignMode == 1 ? inviteEmail : nil,
                inviteeName: assignMode == 1 ? inviteName : nil
            )
            onDone(result.invite?.inviteUrl)
            dismiss()
        } catch {
            errorMessage = error.localizedDescription
        }
    }
}

private struct DomainAdminInviteView: View {
    let mailboxId: String
    var onDone: (String) -> Void

    @Environment(\.dismiss) private var dismiss
    @State private var inviteEmail = ""
    @State private var inviteName = ""
    @State private var asOwner = true
    @State private var isSaving = false
    @State private var errorMessage: String?

    var body: some View {
        Form {
            Section {
                Text(mailboxId)
                    .font(.inter(size: 14))
                    .foregroundStyle(AppTheme.muted)
                TextField("Invitee email", text: $inviteEmail)
                    .textInputAutocapitalization(.never)
                    .keyboardType(.emailAddress)
                TextField("Name (optional)", text: $inviteName)
                Toggle("Invite as owner", isOn: $asOwner)
                    .tint(AppTheme.accent)
            }
            if let errorMessage {
                Section {
                    Text(errorMessage).foregroundStyle(AppTheme.deepDarkRed)
                }
            }
        }
        .navigationTitle("Invite")
        .navigationBarTitleDisplayMode(.inline)
        .toolbar {
            ToolbarItem(placement: .cancellationAction) {
                Button("Cancel") { dismiss() }
            }
            ToolbarItem(placement: .confirmationAction) {
                Button("Send") {
                    Task { await send() }
                }
                .disabled(isSaving || inviteEmail.isEmpty)
            }
        }
    }

    private func send() async {
        isSaving = true
        defer { isSaving = false }
        do {
            let result = try await APIClient.shared.createAdminInvite(
                mailboxId: mailboxId,
                inviteEmail: inviteEmail,
                inviteeName: inviteName.isEmpty ? nil : inviteName,
                role: asOwner ? "owner" : "member"
            )
            onDone(result.inviteUrl)
            dismiss()
        } catch {
            errorMessage = error.localizedDescription
        }
    }
}
